includes/class-sensei-grading.php

```diff
@@ -1131,6 +1131,33 @@ public static function get_graded_lessons_sum() {
 
 	}
 
+	/**
+	 * Get average grade of all lessons graded in all the courses.
+	 *
+	 * @since 4.2.0
+	 * @access public
+	 * @return double $graded_lesson_average_grade Average value of all the graded lessons in all the courses.
+	 */
+	public function get_graded_lessons_average_grade() {
+		// Fetching all the grades of all the lessons that are graded.
+		global $wpdb;
+		$comment_query_piece           = [];
+		$comment_query_piece['select'] = "SELECT SUM( {$wpdb->commentmeta}.meta_value ) AS grade_sum, COUNT( * ) as grade_count";
+		$comment_query_piece['from']   = " FROM {$wpdb->comments}  INNER JOIN {$wpdb->commentmeta}  ON ( {$wpdb->comments}.comment_ID = {$wpdb->commentmeta}.comment_id ) ";
+		$comment_query_piece['where']  = " WHERE {$wpdb->comments}.comment_type IN ('sensei_lesson_status') AND ( {$wpdb->commentmeta}.meta_key = 'grade')";
+
+		$comment_query = $comment_query_piece['select'] . $comment_query_piece['from'] . $comment_query_piece['where'];
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.PreparedSQL.NotPrepared, WordPress.DB.DirectDatabaseQuery.NoCaching -- Performance improvement.
```

Comment:
I don't think you need the `WordPress.DB.PreparedSQL.NotPrepared` anymore since you're not calling `$wpdb->prepare`. Or do you still get a linter error after removing it?

Reply:
I do get a linter error because I'm not using it. Here is the screenshot after I remove it. The same error is displayed when I run it in the terminal. 
<img width="708" alt="image" src="https://user-images.githubusercontent.com/7208249/157874006-f97e41af-7c52-4d68-ad5d-88af7aefe564.png">

Reply:
To avoid this, you should remove the `$comment_query` variable and inline the query string directly in `get_row`.

p1646422816491439-slack-C02NWDZBL0H

Reply:
`$wpdb->get_row( $comment_query )` -> `$wpdb->get_row( "SELECT..." )`

Reply:
@m1r0 [here](https://github.com/Automattic/sensei/pull/4865/commits/59e7db610d025c9a9e294d9e04539592ff702836)